### PR TITLE
[merged] compat: logging.config.dictConfig for < 2.7 added.

### PR DESCRIPTION
**Pull request by ashcrow**
We are using logging.config.dictConfig to allow easy configuration of our loggers via configuration files. However, this functionality was added in 2.7. This change adds a basic implementation of logging.config.dictConfig to allow configuration on 2.6. It is important to note that our reimplementation is not nearly as exhaustive as the true 2.7+ implementation.

### CODE REVIEW

File: src/commissaire/compat/__init__.py [hunk 1]
```diff
@@ -19,4 +19,7 @@
 import platform
 
 #: Version of the current interpreter
-__python_version__ = platform.python_version()[0]
+__python_semver__ = platform.python_version().split('.')
```
Comment (mbarnes) [inline src/commissaire/compat/__init__.py]: A tuple of integers would be a little more convenient:
```
__python_semver__ = tuple(int(x) for x in platform.python_version().split('.'))
```
That's assuming the version string never contains weird non-numeric vendor suffixes.  Seems safe from what the docs say.

File: src/commissaire/compat/logger.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            del kwargs['class']  # class isn't an accepted kwarg
+
+            # Formatter isn't an accepted kwarg
+            formatter_name = kwargs.pop('formatter', None)
+            # Stream isn't an accepted kwarg
+            kwargs.pop('stream', None)
+
+            level_name = 'NOTSET'
+            if kwargs.get('level'):
+                level_name = kwargs.pop('level')
```
Comment (mbarnes) [inline src/commissaire/compat/logger.py]: Could also write this `level_name = kwargs.pop('level', 'NOTSET')`

Comment (ashcrow) [inline src/commissaire/compat/logger.py]: :+1: I like that.

File: src/commissaire/compat/__init__.py (hunk 1 above)
Comment (ashcrow) [inline src/commissaire/compat/__init__.py]: I don't think it should. But then again it's possible someone could patch the platform...

**Comment by mbarnes** [comment]
Would you mind noting in `README` or elsewhere in the repo why we're stuck with Python 2.6 support?  I've already forgotten.  It's for RHEL, right?

Otherwise LGTM with or without my suggestion.  :+1:

**Comment by ashcrow** [comment]
Sure. Yup.

**Comment by ashcrow** [comment]
:arrow_up:

**Comment by mbarnes** [comment]
@rh-atomic-bot try

**Comment by mbarnes** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**